### PR TITLE
perf(service-api): Speed up system.api calls

### PR DESCRIPTION
**Pull request by lucasvanhalst**
The android/ios app now get the option to ask for the service api callback to run synchronously, meaning it will execute the callback immediately and return the result if it was successful. On top of that they should use the 'instant' rpc handler for minimum latency.
Quick testing has shown that for simple calls, latency goes down from 500ms to 50ms.

TODO: test error path, when an error happens the request hangs until it times out (like when you send an incorrect method)

https://github.com/our-city-app/mobicage-android-client/pull/206
https://github.com/our-city-app/mobicage-ios-client/pull/174
https://github.com/our-city-app/oca-backend/pull/1574
https://github.com/our-city-app/cordova-mobicage-plugin/pull/7

After deployment, update the js embedding via mobiadmin/js_embedding

### CODE REVIEW

File: src/rogerthat/bizz/js_embedding/rogerthat/rogerthat.api-1.0.js [hunk 1]
```diff
@@ -51,24 +51,31 @@ var _createRogerthatApiLib = function() {
...
+                // Fast path by default
+                synchronous = true;
+            }
             
             var tagJSONStr = JSON.stringify(tag);
 
             try {
                 if (rogerthat.system && rogerthat.system.os === "android") {
-                    __rogerthat__.sendApiCall(method, params, tagJSONStr);
+                    __rogerthat__.sendApiCall(method, params, tagJSONStr, synchronous);
```
Comment (RubenMattan) [inline src/rogerthat/bizz/js_embedding/rogerthat/rogerthat.api-1.0.js:71]: will this keep working for old clients?

File: src/rogerthat/bizz/service/__init__.py [hunk 2]
```diff
@@ -2594,49 +2594,69 @@ def trans():
...
+                context.put()
+            else:
+                _send_api_call_result_request(app_user, id_, None, None)
 
     xg_on = db.create_transaction_options(xg=True)
-    db.run_in_transaction_options(xg_on, trans)
+    # Since deeper in the stack submit_service_api_callback has a @run_after_transaction decorator,
+    # we cannot run this in a transaction here (when synchronous is True) because that decorator
+    # makes the function sort off async, which means it will always return None.
+    # On the other hand no, transaction means it will run faster so that's nice.
```
Comment (RubenMattan) [inline src/rogerthat/bizz/service/__init__.py:2658]: remove , in comment

File: src/rogerthat/bizz/js_embedding/rogerthat/rogerthat.api-1.0.js (hunk 1 above)
Comment (RubenMattan) [reply src/rogerthat/bizz/js_embedding/rogerthat/rogerthat.api-1.0.js:71]: Just tested and this will nog work for old clients

Comment (lucasvanhalst) [reply src/rogerthat/bizz/js_embedding/rogerthat/rogerthat.api-1.0.js:71]: hm, how do you suggest that we fix that?

**Pull request merged**